client/src/index.tsx

```diff
@@ -1,45 +1,20 @@
 import React from 'react'
 import ReactDOM from 'react-dom/client'
-import { ApolloClient, ApolloProvider, HttpLink, InMemoryCache, split } from '@apollo/client'
-import { GraphQLWsLink } from '@apollo/client/link/subscriptions'
-import { createClient } from 'graphql-ws'
 
 import './index.css'
 import App from './App'
 import reportWebVitals from './reportWebVitals'
-import { getMainDefinition } from '@apollo/client/utilities'
 
-const httpLink = new HttpLink({
-  uri: process.env.REACT_APP_GRAPHQL_API_URL,
-})
-
-const wsLink = new GraphQLWsLink(
-  createClient({
-    url: process.env.REACT_APP_GRAPHQL_API_WS,
-  }),
-)
-
-const splitLink = split(
-  ({ query }) => {
-    const definition = getMainDefinition(query)
-    return definition.kind === 'OperationDefinition' && definition.operation === 'subscription'
-  },
-  wsLink,
-  httpLink,
-)
-
-const client = new ApolloClient({
-  link: splitLink,
-  cache: new InMemoryCache(),
-})
+// common
+import { DomainProvider } from 'common/providers/DomainProvider'
 
 const root = ReactDOM.createRoot(document.getElementById('root') as HTMLElement)
 
 root.render(
   <React.StrictMode>
-    <ApolloProvider client={client}>
+    <DomainProvider>
```

Comment:
Maybe it will be cleaner to create a custom Apollo client context, which will wrap the default `ApolloProvider` and do all the magic to create a client instance within it. This way `App.tsx` will only have routes and provider stuff will be abstracted away from it